pkg/parser/icon_test.go

```diff
@@ -205,6 +204,118 @@ var _ = Describe("icons", func() {
 				Expect(ParseDraftDocument(source)).To(MatchDraftDocument(expected))
 			})
 
+			It("inline icon in title works", func() {
+				source := `== a icon:note[] from me`
+				expected := types.DraftDocument{
+					Blocks: []interface{}{
+						types.Section{
+							Level: 1,
+							Title: []interface{}{
+								types.StringElement{
+									Content: "a ",
+								},
+								types.Icon{
+									Class: "note",
+								},
+								types.StringElement{
+									Content: " from me",
+								},
+							},
+							Elements: []interface{}{},
+						},
+					},
+				}
+				Expect(ParseDraftDocument(source)).To(MatchDraftDocument(expected))
+			})
+
+			It("inline icon at title start", func() {
+				source := `= icon:warning[] or what icon:note[] to do`
+				expected := types.DraftDocument{
+					Blocks: []interface{}{
+						types.Section{
+							Level: 0,
+							Title: []interface{}{
+								types.Icon{Class: "warning"},
+								types.StringElement{Content: " or what "},
+								types.Icon{Class: "note"},
+								types.StringElement{Content: " to do"},
+							},
+							Elements: []interface{}{},
+						},
+					},
+				}
+				Expect(ParseDraftDocument(source)).To(MatchDraftDocument(expected))
+			})
+
+			// NB: The existing grammar for labeled list items does not support any markup
+			// in the term text.
+			It("inline icon as labeled list item description", func() {
+				source := `discount:: icon:tags[alt="Discount"] Cheap cheap!
+item 2:: two`
+				expected := types.DraftDocument{
+					Blocks: []interface{}{
+
+						types.LabeledListItem{
+							Level: 1,
+							Term: []interface{}{
+								types.StringElement{Content: "discount"},
+							},
+							Elements: []interface{}{
+								types.Paragraph{
+									Lines: [][]interface{}{
+										{
+											types.Icon{Class: "tags", Attributes: types.Attributes{types.AttrImageAlt: "Discount"}},
+											types.StringElement{Content: " Cheap cheap!"},
+										},
+									},
+								},
+							},
+						},
+						types.LabeledListItem{
+							Level: 1,
+							Term: []interface{}{
+								types.StringElement{Content: "item 2"},
+							},
+							Elements: []interface{}{
+								types.Paragraph{
+									Lines: [][]interface{}{
+										{
+											types.StringElement{Content: "two"},
+										},
+									},
+								},
+							},
+						},
+					},
+				}
+				Expect(ParseDraftDocument(source)).To(MatchDraftDocument(expected))
+
+			})
+
+			It("inline icon in quoted text", func() {
+				source := `an _italicized icon:warning[] message_`
```

Comment:
sorry to insist, but could you add a test for each kind of quoted text? (bold, monospace, etc.)

Reply:
Ugh.  Ok.  The parser tests are a royal pain.  I have come to really despise the test framework -- it is almost impossible to read the output when there is an error sometimes.

Reply:
you can try something like this:
```
ginkgo -focus "my test" pkg/parser -- -debug
```
with the `-debug` flag passed to the test, you'll have a dump of the actual result, so it's sometimes easier to compare with the expectation. Sometimes the diff is enough, but not always